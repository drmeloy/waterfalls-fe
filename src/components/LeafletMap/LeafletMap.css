.map { 
  height: 400px;
  width: 75%;
  border: 1px solid;
}

