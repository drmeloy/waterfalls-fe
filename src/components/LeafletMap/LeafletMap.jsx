import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Map, Marker, Popup, TileLayer } from 'react-leaflet';
import { Icon } from 'leaflet';

export default function LeafletMap({ waterfalls }){
  const [activeWaterfall, setActiveWaterfall] = useState(null);

  const icon = new Icon({
    iconUrl: '../../public/assets/waterfall-icon.png',
    iconSize: [35, 35]
  });

  const markers = waterfalls.map(waterfall => (
    <Marker 
      key={waterfall.id}
      position={[waterfall.latitude, waterfall.longitude]}
      onClick={() => setActiveWaterfall(waterfall)}
      icon={icon} />
  ));

  const mapStyles = {
    'height': '600px',
    'width': '90vw',
    'margin': '7% auto',
    'margin-bottom': '2%',
    'border': '1px solid'
  };

  return (
    <Map center={[44.1, -120.6]} zoom={7} style={mapStyles}>
      {markers}
      {activeWaterfall && (
        <Popup
          position={[activeWaterfall.latitude, activeWaterfall.longitude]}
          onClose={() => setActiveWaterfall(null)}>
          <div>
            <h6>{activeWaterfall.name}</h6>
            <p>{activeWaterfall.description}</p>
            <span>Height: {activeWaterfall.height}</span><br/>
            <span>Latitude: {activeWaterfall.latitude}</span><br/>
            <span>Longitude: {activeWaterfall.longitude}</span>
          </div>
        </Popup>
      )}
      <TileLayer url='https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png'
        attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors' />
    </Map>
  );
}

LeafletMap.propTypes = {
  waterfalls: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    latitude: PropTypes.string.isRequired,
    longitude: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  }))
};
