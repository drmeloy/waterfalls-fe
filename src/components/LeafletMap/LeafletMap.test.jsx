import React from 'react';
import { shallow } from 'enzyme';
import LeafletMap from './LeafletMap';

describe('LeafletMap component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<LeafletMap waterfalls={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
