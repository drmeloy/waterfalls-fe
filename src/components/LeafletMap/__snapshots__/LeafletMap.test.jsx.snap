// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeafletMap component matches snapshot 1`] = `
<Map
  center={
    Array [
      44.1,
      -120.6,
    ]
  }
  style={
    Object {
      "border": "1px solid",
      "height": "600px",
      "margin": "7% auto",
      "marginBottom": "2%",
      "width": "90vw",
    }
  }
  zoom={7}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="Â© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
  />
</Map>
`;
